xml/src/main/scala/akka/stream/alpakka/xml/model.scala

```diff
@@ -48,6 +48,11 @@ private class MapOverTraversable[A, K, V](source: Traversable[A], fKey: A => K,
 
   override def iterator: Iterator[(K, V)] = source.toIterator.map(a => (fKey(a), fValue(a)))
 
+  def remove(key: K): scala.collection.immutable.Map[K, V] =
+    throw new UnsupportedOperationException("Convert to map from iterator to support removing elements.")
```

Comment:
`Traversable` is not really an iterator, since it should be traversable multiple times (otherwise it'd only be a `TraversableOnce`) - couldn't we do something like `new MapOverTraversable(source.filter(k => fKey(k) != key), fKey, fValue)` here?